packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -211,13 +236,53 @@ export function useConnections({
   );
   const { activeConnectionId } = state;
 
+  const getActiveAndInactiveCount = useCallback(() => {
+    const total = favoriteConnections.length + recentConnections.length;
+    const active =
+      connectionsManager.getConnectionIdsByStatus(ConnectionStatus.Connected)
+        .length || 0;
+    return { active, inactive: total - active };
+  }, [favoriteConnections, recentConnections, connectionsManager]);
+
+  const findConnectionInfo = useCallback(
+    (connectionId: ConnectionInfo['id']) => {
+      return [...favoriteConnections, ...recentConnections].find(
+        ({ id }) => id === connectionId
+      );
+    },
+    [favoriteConnections, recentConnections]
+  );
+
   const saveConnectionInfo = useCallback(
-    async (
-      connectionInfo: RecursivePartial<ConnectionInfo> &
-        Pick<ConnectionInfo, 'id'>
-    ) => {
+    async ({
+      fullConnectionInfo,
+      partialConnectionInfo,
+    }:
+      | {
+          fullConnectionInfo?: undefined;
+          partialConnectionInfo: RecursivePartial<ConnectionInfo> &
+            Pick<ConnectionInfo, 'id'>;
+        }
+      | {
+          fullConnectionInfo: ConnectionInfo;
+          partialConnectionInfo?: undefined;
+        }) => {
+      const connectionInfo = fullConnectionInfo || partialConnectionInfo;
       try {
-        await saveConnection(connectionInfo);
+        if (fullConnectionInfo) {
+          const isNewConnection = !findConnectionInfo(fullConnectionInfo.id);
+          if (isNewConnection) {
+            const { active, inactive } = getActiveAndInactiveCount();
+            const isActive =
+              connectionsManager.statusOf(fullConnectionInfo.id) ===
+              'connected'; // save normally happens first, but in case.
+            onConnectionCreated?.(fullConnectionInfo, {
+              active: isActive ? active + 1 : active,
+              inactive: isActive ? inactive : inactive + 1, // we can't use ConnectionRepository here, as it updates asynchronously
+            });
+          }
+        }
+        await repositorySaveConnection(connectionInfo || partialConnectionInfo);
```

Comment:
`connectionInfo || partialConnectionInfo`

We already have
`const connectionInfo = fullConnectionInfo || partialConnectionInfo;` few lines above?

Reply:
thanks, I forgot to update this line!